.husky/pre-commit

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+# Run all hooks inside pre-commit.d
+node "$(dirname "$0")/pre-commit.d/check-assert.js"
```

Comment:
Have you considered making this pre-push rather than pre-commit? I commonly will save work-in-progress by committing it, where I would not expect it to pass any linters or whatnot. We opted for pre-push in the core monorepo.

Reply:
I hadn't, but I still think pre-commit hooks work better here because the purpose of the hook is to shorten the feedback loop as much as possible.

For specific rules in the future, we should be able to customize if we want them enforced in the CI on pre-push or pre-commit hooks (opt-in per your suggestion, giving devs freedom to express their preferences) on a per-rule basis. It'll become more clear how that works once we introduce the second rule.